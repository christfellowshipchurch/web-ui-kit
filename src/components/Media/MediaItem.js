import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import Image from './Image';
import Video from './Video';

import {
  mediaContainer1by1, mediaContainer4by3, mediaContainer16by9, mediaContainer21by9,
  mediaContentContainer, round
} from '../css/styles.module.css'

const MediaItem = ({
  ratio, imageUrl, imageAlt, videoUrl, className, children, rounded
}) => {
  const ratioClasses = {
    '1by1': mediaContainer1by1,
    '4by3': mediaContainer4by3,
    '16by9': mediaContainer16by9,
    '21by9': mediaContainer21by9,
  }

  const classes = classNames(
    ratioClasses[ratio],
    rounded ? round : '',
    className
  )

  return (
    <div className={classes}>
      <Image source={imageUrl} alt={imageAlt} />
      {videoUrl
        ? <Video source={videoUrl} />
        : null}

      {children
        ? <div className={mediaContentContainer}>{children}</div>
        : null}
    </div>
  )
};

const defaultProps = {
  ratio: '1by1',
  videoUrl: null,
  className: ''
}

const propTypes = {
  ratio: PropTypes.string,
  imageUrl: PropTypes.string.isRequired,
  imageAlt: PropTypes.string.isRequired,
  videoUrl: PropTypes.string,
  className: PropTypes.string,
}

MediaItem.defaultProps = defaultProps;
MediaItem.propTypes = propTypes;

export default MediaItem;
