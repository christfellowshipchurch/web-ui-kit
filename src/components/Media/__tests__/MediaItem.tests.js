import React from 'react'
import MediaItem from '../../Media'
import renderer from 'react-test-renderer'

describe('MediaItem', () => {
    it('renders an image with no video', () => {
        const tree = renderer.create(
            <MediaItem imageUrl="https://img.url.com" imageAlt="Image Alt" />
        )

        expect(tree).toMatchSnapshot()
    })

    it('renders a video with a background image', () => {
        const tree = renderer.create(
            <MediaItem videoUrl="https://video.url.com" imageUrl="https://img.url.com" imageAlt="Image Alt" />
        )

        expect(tree).toMatchSnapshot()
    })

    it('renders content overtop of an image', () => {
        const tree = renderer.create(
            <MediaItem imageUrl="https://img.url.com" imageAlt="Image Alt">
                <h1>This is content overtop of an image</h1>
            </MediaItem>
        )

        expect(tree).toMatchSnapshot()
    })

    it('renders a rounded media item', () => {
        const tree = renderer.create(
            <MediaItem imageUrl="https://img.url.com" imageAlt="Image Alt" rounded />
        )

        expect(tree).toMatchSnapshot()
    })
})