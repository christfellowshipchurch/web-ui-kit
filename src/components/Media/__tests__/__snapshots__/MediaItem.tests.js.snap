// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaItem renders a rounded media item 1`] = `
<div
  className=""
>
  <div>
    <div />
  </div>
</div>
`;

exports[`MediaItem renders a video with a background image 1`] = `
<div
  className=""
>
  <div>
    <div />
  </div>
  <video
    autoPlay={true}
    loop={true}
    muted={true}
    playsInline={true}
  >
    <source
      src="https://video.url.com"
      type="video/mp4"
    />
  </video>
</div>
`;

exports[`MediaItem renders an image with no video 1`] = `
<div
  className=""
>
  <div>
    <div />
  </div>
</div>
`;

exports[`MediaItem renders content overtop of an image 1`] = `
<div
  className=""
>
  <div>
    <div />
  </div>
  <div>
    <h1>
      This is content overtop of an image
    </h1>
  </div>
</div>
`;
